go/vt/vttest/environment.go

```diff
@@ -144,9 +147,14 @@ func (env *LocalTestEnv) BinaryPath(binary string) string {
 
 // MySQLManager implements MySQLManager for LocalTestEnv
 func (env *LocalTestEnv) MySQLManager(mycnf []string, snapshot string) (MySQLManager, error) {
+	// maria db doesn't recognize super-read-only, therefore we have separate sql for that.
+	var initFile = path.Join(os.Getenv("VTROOT"), "config/init_db.sql")
+	if !env.IsSQLFlavor() {
+		initFile = path.Join(os.Getenv("VTROOT"), "config/init_maria_db.sql")
+	}
```

Comment:
Instead of asking the environment, we could query the MySQL instance directly to know if it is mariadb distribution or not. This would prevent users from running into configuration issues.